citrination_client/base/response_handling.py

```diff
@@ -0,0 +1,73 @@
+from citrination_client.base.errors import *
+from time import sleep
+
+def check_for_rate_limiting(response, response_lambda, timeout=1, attempts=0):
+    """
+    Takes an initial response, and a way to repeat the request that produced it and retries the request with an increasing sleep period between requests if rate limiting resposne codes are encountered.
```

Comment:
Can you set up a style checker as a part of this PR. It's going to hurt, but this would be a great time to do it.

Reply:
I disagree.  This PR contains real behavioral changes, whereas compliance with a style checker is mostly formatting.  I think it would be better to keep them in two separate PRs.  Remember, this is into `develop`, not a release, so we aren't pushing major code changes to users twice...